core/src/main/scala/cats/data/Continuation.scala

```diff
@@ -0,0 +1,108 @@
+package cats
+package data
+
+import java.io.Serializable
+import scala.annotation.tailrec
+
+sealed abstract class Continuation[O, +I] extends Serializable {
```

Comment:
I was also going to suggest a signature of `Continuation[+I, O]`, unless there's some reason for the current order.   Is there a reason it's `O` and not `-O` ?